### PR TITLE
Feature/user view

### LINKED ISSUE #40: Format state in the Banner (especially for Yellowstone)

### LINKED ISSUE #41: Fix Randomize button in MainNav Component

### PR DESCRIPTION
**Pull request by baileydunning**
## Is this a fix or a feature?
Feature

## What is the change?
+ Adds functionality for user to be able to navigate to About, Location and Contact for each individual park
+ Allows user to add and remove parks to their Saved List (either Visited or Bucket List)
+ Add styles
+ Fix a lot of TypeScript bugs

## Where should the reviewer start?
+ `ParkContainer` & children `ParkInfo` & `SavedParks`
+ `App`

## How should this be tested?
+ `cd` into `national-parkfinder` > `npm install` > `npm start`
+ Open application in browser on `localhost:3000`

## Relevant Tickets
+ Closes #40 
+ Closes #41
+ #9 
+ #11
+ #5 
+ #7 
+ #1 
+ #37 

### Gif

@saraho1123
@jaypeasee
@baileydunning

### CODE REVIEW

File: src/Banner/Banner.tsx [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
+  const randomIndex = Math.floor(Math.random() * images.length)
+
+  return (
+    <section className='banner'>
+      <div className='header-container'>
+        <h1 className='header'>{currentPark.fullName}, {currentPark.states}</h1>
+        <p className='caption'><i>{images[randomIndex].caption}</i></p>
+      </div>
+      <div>
+        {<img
```
Comment (jaypeasee) [inline src/Banner/Banner.tsx:19]: I like that we're now randomizing these images. Alot of them at index 0 were kind of lame and not representational of the park.

File: src/BannerIcons/BannerIcons.tsx [hunk 2]
```diff
@@ -1,30 +1,85 @@
...
-                <h2>Add To Bucket List</h2>
-            </div>
-        </section>
-    )
+type BannerIconsProps = AddRemoveFunctionality | { localPark: () => any | void } 
+
+
+const BannerIcons: React.FC<BannerIconsProps> = props => {
+  const { addToVisited } = props as AddRemoveFunctionality
+  const { deleteFromVisited } = props as AddRemoveFunctionality
```
Comment (jaypeasee) [inline src/BannerIcons/BannerIcons.tsx:14]: The interface for this was crucial to this entire PR

File: src/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Link } from 'react-router-dom'
+import './Footer.scss'
+
+// pass in buttons as props to generate links for footer
```
Comment (saraho1123) [inline src/Footer/Footer.tsx:4]: Was this a note from Kaitlin?

File: src/Footer/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+const Footer: React.FC = () => {
```
Comment (saraho1123) [inline src/Footer/Footer.tsx:6]: I like that these have been moved to the footer!

File: src/SavedCard/SavedCard.tsx [hunk 5]
```diff
@@ -1 +1,44 @@
...
+        src={image}
+        alt={`photograph of ${name}`}
+        className="saved-park-img"
+      />
+      <h2>{name}, {state}</h2>
+      {location.pathname === '/user/visited' && <button
+        onClick={() => deleteFromVisited(parkCode)}>
+        Delete from Visited
+      </button>}
+      {location.pathname === '/user/bucket-list' && <button
```
Comment (jaypeasee) [inline src/SavedCard/SavedCard.tsx:31]: Great way to use `useLocation()` for conditionally rendering buttons.

File: src/NavSearch/NavSearch.tsx [hunk 6]
```diff
@@ -2,29 +2,28 @@ import React, { useState } from 'react'
 import './NavSearch.scss'
 
 interface FilterButtonsByName {
-    filterButtonsByName: (searchTerm: string) => void
+  filterButtonsByName: (searchTerm: string) => void
```
Comment (saraho1123) [inline src/NavSearch/NavSearch.tsx:5]: Nice teamwork getting this working!

File: src/SavedParks/SavedParks.tsx [hunk 7]
```diff
@@ -1,7 +1,42 @@
-const SavedParks: React.FC = () => {
+import { useEffect, useState } from 'react'
+import { LocalParkContainer, LocalParkData, AddRemoveFunctionality } from '../interfaces'
+import SavedCard from '../SavedCard/SavedCard'
+import { useLocation } from 'react-router-dom'
+
+const SavedParks: React.FC<LocalParkContainer & AddRemoveFunctionality> = props => {
```
Comment (jaypeasee) [inline src/SavedParks/SavedParks.tsx:6]: Nice solution for allowing us to pass multiple props with different interfaces to `SavedCard` component

File: src/Footer/Footer.tsx (hunk 3 above)
Comment (baileydunning) [reply src/Footer/Footer.tsx:4]: Yeah it is. She was suggesting we make separate components to render in Footer

File: src/ParkContainer/ParkContainer.tsx [hunk 8]
```diff
@@ -3,54 +3,121 @@ import Banner from '../Banner/Banner'
...
+  const findChosenPark = (parkCode: string): any | void => {
+    const chosenPark = nationalParks.find(park => {
+      return park.parkCode === parkCode
+    })
+    if (chosenPark) {
+      return chosenPark
+    }
+  }
+
+  const addToVisited = (parkCode: string) => {
```
Comment (saraho1123) [inline src/ParkContainer/ParkContainer.tsx:40]: Another great moment of teamwork here!

**Review by jaypeasee — APPROVED** [review]
Great work! MVP is made, now we gotta test everything.......

File: src/ParkInfo/ParkInfo.tsx [hunk 9]
```diff
@@ -2,16 +2,22 @@ import './ParkInfo.scss'
...
 import Location from '../Location/Location'
 import Contact from '../Contact/Contact'
+import BannerIcons from '../BannerIcons/BannerIcons'
 import { Switch, Route } from 'react-router-dom'
-import { CurrentParkContainer } from '../interfaces'
-import { RouteComponentProps } from 'react-router-dom'
+import { AddRemoveFunctionality, CurrentParkContainer } from '../interfaces'
 
-type ParkInfoProps = CurrentParkContainer | RouteComponentProps
+type ParkInfoProps = CurrentParkContainer | AddRemoveFunctionality | { localPark: () => any | void}
```
Comment (saraho1123) [inline src/ParkInfo/ParkInfo.tsx:9]: This is cool. I didn't realize that we can assign a type in here without it being an interface

File: src/SavedParks/SavedParks.tsx (hunk 7 above)
Comment (saraho1123) [inline src/SavedParks/SavedParks.tsx:6]: Nice work figuring this one out with you Bailey!

File: src/interfaces.tsx [hunk 10]
```diff
@@ -82,4 +87,11 @@ export interface CurrentPark {
 
 export interface CurrentParkContainer {
   currentPark: CurrentPark
+}
+
+export interface AddRemoveFunctionality {
```
Comment (saraho1123) [inline src/interfaces.tsx:92]: Glad to know we can add methods to the interface!

**Review by saraho1123 — APPROVED** [review]
We got our functionality done! Great job on all that yesterday!

**Pull request merged**